### ISSUE TITLE
How to achieve better generalization from diverse source inputs?

### ISSUE THREAD

**Issue by moiseshorta** [marcoppasini/MelGAN-VC#16]
Dear Marco,

Wishing you a happy new year.

I've been working extensively with MelGAN-VC in my practice the better part of last year.

I have recently run into the issue that the model is a bit unpredictable when trying to generalize on unseen source sound inputs. Generally speaking, when performing VC on unseen source audio, the model sound synthesis is not so reminiscent of the target audio, only in some cases which I cannot determine.

Any tips on overcoming this? Thanks!

**Comment by DavidTxx** [comment]
That may be because of overfitting. Have you tried adding regularization techniques in the model?

**Comment by moiseshorta** [comment]
@DavidTxx By regularization techniques, you mean augmenting the dataset? if this is not what you mean, I'm curious to know which approach you took, thanks.

**Comment by DavidTxx** [comment]
Potentially, but I was rather thinking about Dropout, Early Stopping, Learning Rate Schedule, ... Experimentation is the best anyway.

**Comment by moiseshorta** [comment]
@DavidTxx Thanks for pointing in that direction. not really an engineer here, so will look up those techniques. Any good resource on where to learn about them better? thanks

**Comment by DavidTxx** [comment]
Read books and docs, which take time -  
https://mitpress.mit.edu/books/deep-learning
https://keras.io/guides/
